CMakeLists.txt

```diff
@@ -70,10 +70,16 @@ endif()
 # https://cmake.org/cmake/help/v3.12/module/CPack.html
 # https://cmake.org/cmake/help/v3.7/module/CPackRPM.html
 include(GNUInstallDirs)
-set(CMAKE_INSTALL_PREFIX "/usr")
+
+set(SHARED_CONFIG_PATH "${CMAKE_INSTALL_FULL_DATAROOTDIR}/touchegg/touchegg.conf")
+target_compile_definitions(touchegg PUBLIC SHARED_CONFIG_PATH=\"${SHARED_CONFIG_PATH}\")
+
+# configure systemd service unit to use the right path, e.g. ${CMAKE_INSTALL_BINDIR}/touchegg
+configure_file(${PROJECT_SOURCE_DIR}/installation/touchegg.service.in ${PROJECT_SOURCE_DIR}/installation/touchegg.service)
+
 install(FILES ${PROJECT_SOURCE_DIR}/installation/touchegg.conf DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/touchegg) # /usr/share/touchegg/touchegg.conf
-install(FILES ${PROJECT_SOURCE_DIR}/installation/touchegg.service DESTINATION /lib/systemd/system)
```

Comment:
Maybe CPack below should be updated as well but I have no idea about CPack.

Reply:
I am not sure I understand. For me CMake builds correctly and installs the service file. Can you expand on that?

Reply:
I meant to attach this to line 109 and below but GitHub does not allow that. But I have no idea how CPack works so I only wanted to highlight the hardcoded paths so that someone familiar with CPack could double check that.

Reply:
I think that as Jan suggested [here](https://github.com/JoseExposito/touchegg/pull/467#issuecomment-791968792), we could use [pkg_get_variable](https://cmake.org/cmake/help/latest/module/FindPkgConfig.html#command:pkg_get_variable) to get this path.
Something like `pkg_get_variable(SYSTEMD_SERVICE_DIR systemd systemdsystemunitdir)` should work.

CPack is used to generate the RPM package. The same variable should be used in lines 109 to 101.

Reply:
Something like this?
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1651895..5c24849 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,9 +76,10 @@ target_compile_definitions(touchegg PUBLIC SYSTEM_CONFIG_FILE_PATH=\"${SYSTEM_CO
 
 # configure systemd service unit to use the right path, e.g. @CMAKE_INSTALL_BINDIR@/touchegg
 configure_file(${PROJECT_SOURCE_DIR}/installation/touchegg.service.in ${PROJECT_SOURCE_DIR}/installation/touchegg.service @ONLY)
+pkg_get_variable(SYSTEMD_SERVICE_DIR systemd systemdsystemunitdir)
 
 install(FILES ${PROJECT_SOURCE_DIR}/installation/touchegg.conf DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/touchegg) # /usr/share/touchegg/touchegg.conf
-install(FILES ${PROJECT_SOURCE_DIR}/installation/touchegg.service DESTINATION ${CMAKE_INSTALL_LIBDIR}/systemd/system)
+install(FILES ${PROJECT_SOURCE_DIR}/installation/touchegg.service DESTINATION ${SYSTEMD_SERVICE_DIR})
 install(FILES ${PROJECT_SOURCE_DIR}/installation/touchegg.desktop DESTINATION ${CMAKE_INSTALL_SYSCONFDIR}/xdg/autostart)
 install(PROGRAMS ${PROJECT_BINARY_DIR}/touchegg DESTINATION ${CMAKE_INSTALL_BINDIR}) # /usr/bin/touchegg
 
@@ -106,9 +107,7 @@ set(CPACK_RPM_POST_INSTALL_SCRIPT_FILE "${PROJECT_SOURCE_DIR}/installation/rpm/p
 list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "${CMAKE_INSTALL_PREFIX}")
 list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "${CMAKE_INSTALL_BINDIR}")
 list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "${CMAKE_INSTALL_DATAROOTDIR}")
-list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/lib")
-list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/lib/systemd")
-list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/lib/systemd/system")
+list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "${SYSTEMD_SERVICE_DIR}")
 list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/etc")
 list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/etc/xdg")
 list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/etc/xdg/autostart")
```
@jtojnar  Do you know how `PKG_CONFIG_SYSTEMD_SYSTEMDSYSTEMUNITDIR` should be set in Nix then? Maybe something like 
  ```nix
preConfigure =''
    PKG_CONFIG_SYSTEMD_SYSTEMDSYSTEMUNITDIR="$out/lib/systemd/system/"
'';
```

Reply:
> @jtojnar Do you know how `PKG_CONFIG_SYSTEMD_SYSTEMDSYSTEMUNITDIR` should be set in Nix then? Maybe something like
> 
> ```nix
> preConfigure =''
>   PKG_CONFIG_SYSTEMD_SYSTEMDSYSTEMUNITDIR="$out/lib/systemd/system/"
> '';
> ```

That should work, or just pass `PKG_CONFIG_SYSTEMD_SYSTEMDSYSTEMUNITDIR = "${placeholder "out"}/lib/systemd/system/";` to `mkDerivation`.

Reply:
> > @jtojnar Do you know how `PKG_CONFIG_SYSTEMD_SYSTEMDSYSTEMUNITDIR` should be set in Nix then? Maybe something like
> > ```nix
> > preConfigure =''
> >   PKG_CONFIG_SYSTEMD_SYSTEMDSYSTEMUNITDIR="$out/lib/systemd/system/"
> > '';
> > ```
> 
> That should work, or just pass `PKG_CONFIG_SYSTEMD_SYSTEMDSYSTEMUNITDIR = "${placeholder "out"}/lib/systemd/system";` to `mkDerivation`.

Even better, thanks!

Reply:
> @JoseExposito Something like this?

The first bit `pkg_get_variable(...)` and the install are correct.

For the `CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION` you will need a loop like:

```diff
-list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/lib")
-list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/lib/systemd")
-list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/lib/systemd/system")
list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/etc")
list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/etc/xdg")
list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/etc/xdg/autostart")
+
+ # Exclude the systemd service dir (usually /lib/systemd/system). Split the path and exclude each subpath.
+ # Equivalent to exclude /lib, /lib/systemd and /lib/systemd/system:
+string(REPLACE "/" ";" SYSTEMD_SERVICE_DIRS ${SYSTEMD_SERVICE_DIR})
+set(EXCLUDE_DIR "")
+foreach(DIR ${SYSTEMD_SERVICE_DIRS})
+  set(EXCLUDE_DIR "${EXCLUDE_DIR}/${DIR}")
+  list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION ${EXCLUDE_DIR})
+endforeach()
```

Reply:
> The first bit `pkg_get_variable(...)` and the install are correct.
> 
> For the `CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION` you will need a loop like:

Okay great! That should be in 62baa28